### PR TITLE
Add HuggingFace TGI publish and test pipeline

### PR DESCRIPTION
**Pull request by xyang16**
## Description ##

Brief description of what this PR is about

Add HuggingFace TGI docker build pipeline

DLC PR: https://github.com/aws/deep-learning-containers/pull/2910

### CODE REVIEW

File: .github/workflows/docker-publish-huggingface.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    timeout-minutes: 180
+    needs: create-runner
+    env:
+      TGI_VERSION: ${{github.event.inputs.tgi-version}}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: huggingface/text-generation-inference
+          ref: v${{ env.TGI_VERSION }}
+      - uses: actions/checkout@v3
```
Comment (lanking520) [inline .github/workflows/docker-publish-huggingface.yml:38]: probably you need to clean up previous build containers. Check how we do in sync job

Comment (xyang16) [reply .github/workflows/docker-publish-huggingface.yml]: Done

File: .github/workflows/docker-publish-hf.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+          repository: aws/deep-learning-containers
+          path: deep-learning-containers
+      - name: Setup Docker buildx
+        uses: docker/setup-buildx-action@v2
+        with:
+          install: true
+      - name: Inject slug/short variables
+        uses: rlespinasse/github-slug-action@v4
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
```
Comment (lanking520) [inline .github/workflows/docker-publish-hf.yml:49]: why we still need this given credential inside?

File: .github/workflows/hf_tgi_integration.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      - name: Create new G5 instance
+        id: create_gpu
+        run: |
+          cd /home/ubuntu/djl_benchmark_script/scripts
+          token=$( curl -X POST -H "Authorization: token ${{ secrets.ACTION_RUNNER_PERSONAL_TOKEN }}" \
+          https://api.github.com/repos/deepjavalibrary/djl-serving/actions/runners/registration-token \
+          --fail \
+          | jq '.token' | tr -d '"' )
+          ./start_instance.sh action_g5 $token djl-serving
+      - name: Create new G5 instance
```
Comment (lanking520) [inline .github/workflows/hf_tgi_integration.yml:24]: why create two G5 instance here?

File: .github/workflows/hf_tgi_integration.yml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      - name: Pull docker
+        env:
+          REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+        run: |
+          docker pull ${REGISTRY}/djl-serving:tgi-${TGI_VERSION}
+      - name: Test bloom-560m
+        env:
+          REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+        run: |
+          HF_MODEL_ID=bigscience/bloom-560m && \
```
Comment (lanking520) [inline .github/workflows/hf_tgi_integration.yml:69]: probably some larger model and more model to tests

File: .github/workflows/docker-publish-hf.yml (hunk 2 above)
Comment (xyang16) [reply .github/workflows/docker-publish-hf.yml:49]: This is only setting up region. See https://github.com/aws-actions/configure-aws-credentials#self-hosted-runners.

File: .github/workflows/hf_tgi_integration.yml (hunk 3 above)
Comment (xyang16) [reply .github/workflows/hf_tgi_integration.yml:24]: Fixed.

File: .github/workflows/hf_tgi_integration.yml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    if: always()
+    runs-on: [ self-hosted, scheduler ]
+    needs: [ create-runners, hf-tgi-test ]
+    steps:
+      - name: Stop all instances
+        run: |
+          cd /home/ubuntu/djl_benchmark_script/scripts
+          instance_id=${{ needs.create-runners.outputs.gpu_instance_id_1 }}
+          ./stop_instance.sh $instance_id
+          instance_id=${{ needs.create-runners.outputs.gpu_instance_id_2 }}
```
Comment (lanking520) [inline .github/workflows/hf_tgi_integration.yml:95]: stop two instances?

Comment (xyang16) [reply .github/workflows/hf_tgi_integration.yml]: Somehow these changes were there in my local and not pushed.

File: .github/workflows/hf_tgi_integration.yml [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+          REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+        run: |
+          HF_MODEL_ID=bigscience/bloom-560m && \
+          SM_NUM_GPUS=2 && \
+          TGI_VERSION=0.5.0 && \
+          docker run --gpus all --shm-size 1g -itd --rm -p 8080:8080 \
+              -e SM_NUM_GPUS=$SM_NUM_GPUS -e HF_MODEL_ID=$HF_MODEL_ID ${REGISTRY}/djl-serving:tgi-${TGI_VERSION}
+          sleep 90
+          ret=$(curl http://localhost:8080/invocations -X POST \
+              -d '{"inputs":"What is Deep Learning?","parameters":{"max_new_tokens":17}}' \
```
Comment (lanking520) [inline .github/workflows/hf_tgi_integration.yml:66]: Let's do 128 max_new_tokens here

**Review by lanking520 — APPROVED** [review]

**Pull request merged**